### ISSUE TITLE
Showing output of external program (ffmpeg) in compiled file

### ISSUE THREAD

**Issue by mbassiouny33** [MScholtes/PS2EXE#90]
Hi there

Thanks a lot for this great tool

I have a ps1 script as follows:

```

# take some parameters process them, do things to find param1 param2 then do
$ErrorActionPreference = 'SilentlyContinue'
& .\ffmpeg.exe param1 param2   #some exe program that output things to stdout
```

the `ffmpeg.exe` may sometimes output errors or interactively ask for permission (press Y or N) to overwrite files. This is ok when I run the script directly in Powershell and everything works great.

But when I compile it to .exe with ps2exe all output (and interactive prompts) from the external program are totally ignored and never appear.

I made sure -no console and -noOutput flags are not set. I'm compiling via the basic command ` ps2exe file.ps1 file.exe`

Could you please help me to make the compiled version of the script pass interaction from the external program?

Thanks

**Comment by MScholtes** [comment]
Hello @mbassiouny33,

I can't tell much about this. I guess **my_external_program.exe** does not use standard output and standard input.
What program is it?

You might see the effect when starting **systeminfo.exe**. The status messages right at the start of program are not given out to standard output and therefore do not appear with **PSEXE**. But the result is handed to standard output and therefore appear with **PS2EXE**.

Greetings

Markus

**Comment by mbassiouny33** [comment]
thanks for your reply @MScholtes 

>  my_external_program.exe does not use standard output and standard input.
What program is it?

I tried several programs to double-check if my program uses std out, and it appears it does not as you have mentioned, the particular program in question here that doesn't work is ffmpeg.exe from https://www.gyan.dev/ffmpeg/builds/
to reproduce it is enough toplace ffmpeg.exe from the archive on the website in the same folder as the script and run something like `ffmpeg.exe -version`

Thanks

**Comment by mbassiouny33** [comment]
Sorry that I bothered you with an issue that turned out to be some weirdness about ffmpeg...

FFmpeg does not use stdout (as you suggested) as these messages are errors.  A partial workaround is to redirect the error to standard output with `2>1` at the end of the command. 
However, I was still not able to figure out a way to get the interactive prompt that asks to press Y/N to continue to appear. nor the progress message of ffmpeg (the message that looks like `size=   29922kB time=00:12:40.04 bitrate= 132.0kbits/s speed=20.5x`) So what I did in the end is 

```
ffmpeg.exe -y -nostdin -progress  -progress pipe:1 ....
```
progress will throw the progress (in a very messy way instead of the usual compact way) on pipe 1 which is the standard output, 
-y and -nostdin will prevent interaction and force overwrite.

Ideally, I would prefer to keep the interaction, but I can't find a way to preserve it...


The issue turned out to be not exactly ps2exe related in the end, but it is just that ps2exe makes ffmpeg's mess more visible.
